ci/src/revisions/git_lib.py

```diff
@@ -0,0 +1,167 @@
+import argparse
+import logging
+import pathlib
+import subprocess
+import sys
+from typing import List
+
+
+class HelpfulParser(argparse.ArgumentParser):
+    """An argparse parser that prints usage on any error."""
+
+    def error(self, message):
+        sys.stderr.write("error: %s\n" % message)
+        self.print_help()
+        sys.exit(2)
+
+
+def init_helpful_parser() -> HelpfulParser:
+    parser = HelpfulParser()
+    parser.add_argument("--verbose", "-v", action="store_true", help="Verbose mode")
+
+    return parser
+
+
+def get_logger(level) -> logging.Logger:
+    FORMAT = "[%(asctime)s] %(levelname)-8s %(message)s"
+    logging.basicConfig(format=FORMAT, level=level)
+    return logging.getLogger("logger")
+
+
+def get_repo_root() -> pathlib.Path:
+    return pathlib.Path(
+        subprocess.run(["git", "rev-parse", "--show-toplevel"], text=True, stdout=subprocess.PIPE).stdout.strip()
+    )
+
+
+def sync_main_branch_and_checkout_branch(
+    repo_root: pathlib.Path, main_branch: str, branch_to_checkout: str, logger: logging.Logger
+):
+    if not repo_root.exists():
+        raise Exception("Expected dir %s to exist", repo_root.name)
+
+    subprocess.call(["git", "fetch", "origin", f"{main_branch}:{main_branch}"], cwd=repo_root)
+
+    result = subprocess.run(["git", "status", "--porcelain"], stdout=subprocess.PIPE, text=True, check=True)
+    if result.stdout.strip():
+        logger.warn("Uncommited work:\n%s", result.stdout.strip())
+        logger.error("Found uncommited work! Commit and then proceed.")
+        exit(2)
+
+    if subprocess.call(["git", "checkout", branch_to_checkout], cwd=repo_root) == 0:
+        # The branch already exists, update the existing MR
+        logger.info("Found an already existing target branch")
```

Comment:
Does this condition ever hold considering the fact that we're only fetching the master branch above (and in the actions/checkout step)? Don't you need to try to fetch the `branch_to_checkout` as well above and deal with the case when it doesn't exist?

Reply:
Oh nevermind. I see below you're always resetting the branch to `master` so it's not really necessary to first fetch the branch from the origin.